easybuild/easyblocks/t/tensorflow.py

```diff
@@ -0,0 +1,308 @@
+##
+# Copyright 2017-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing TensorFlow, implemented as an easyblock
+
+@author: Kenneth Hoste (HPC-UGent)
+@author: Ake Sandgren (Umea University)
+@author: Damian Alvarez (Forschungzentrum Juelich GmbH)
+"""
+import glob
+import os
+import stat
+import tempfile
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, mkdir, resolve_path
+from easybuild.tools.filetools import which, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+# wrapper for Intel compiler, where required environment are hardcoded to make sure they're present;
+# this is required because Bazel resets the environment in which compiler commands are executed...
+INTEL_COMPILER_WRAPPER = """#!/bin/bash
+
+export CPATH='%(cpath)s'
+export INTEL_LICENSE_FILE='%(intel_license_file)s'
+
+%(compiler_path)s "$@"
+"""
+
+
+class EB_TensorFlow(PythonPackage):
+    """Support for building/installing TensorFlow."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'with_jemalloc': [True, "Make TensorFlow use jemalloc", CUSTOM],
```

Comment:
Wouldn't it be better to use jemalloc only when it is present as a dependency and remove this with_jemalloc thingie?

Reply:
Currently we have no way to let Bazel pick up on self-provided dependencies, but we needed a way to disable it pulling in and building `jemalloc` itself (which is required on CentOS 6).

This is something to improve on in the future, but I think we'll have to wait for improvements in Bazel itself before this is feasible without heavy patching, it's a huge PITA right now to make that work.

This easyblock is a starting point, definitely not the end of the road...

Reply:
Ok, good enough for me then.
Awaiting merge so i can redo the builds...